tests/es5.js

```diff
@@ -0,0 +1,202 @@
+      expect(error).to.be.an("error");
+      expect(error.message).to.equal("apple");
+      done();
+    });
+  });
+
+  it("correctly rejects when getBlockByHash throws", function (done) {
+    var getBlockByHash = function (hash, callback) {
+      throw new Error("apple");
+    }
+    var blockHistory = ImmutableList([new MockBlock(0x7777)]);
+    var newBlock = new MockBlock(0x7779);
+    var blockRetention = 5;
+
+    reconcileBlockHistoryEs5(getBlockByHash, blockHistory, newBlock, onBlockAdded, onBlockRemoved, blockRetention, function (error, newHistory) {
+      expect(error).to.be.an("error");
+      expect(error.message).to.equal("apple");
+      done();
+    });
+  });
+
+  it("correctly rejects when onBlockAdded calls back with error", function (done) {
+    var getBlockByHash = function (hash, callback) { throw new Error("unreachable"); }
+    var failingOnBlockAdded = function (block, callback) { callback(new Error("apple")); }
+    var blockHistory = null;
+    var newBlock = new MockBlock(0x7777);
+    var blockRetention = 5;
+
+    reconcileBlockHistoryEs5(getBlockByHash, blockHistory, newBlock, failingOnBlockAdded, onBlockRemoved, blockRetention, function (error, newHistory) {
+      expect(error).to.be.an("error");
+      expect(error.message).to.equal("apple");
+      done();
+    });
+  });
+
+  it("accepts onBlockRemoved function that uses a callback", function (done) {
+    var getBlockByHash = getBlockByHashFactory();
+    var getBlockByHash = function (hash, callback) { throw new Error("unreachable"); }
+    var blockHistory = ImmutableList([
+      new MockBlock(0x7777, "AAAA", "AAAA"),
+      new MockBlock(0x7778, "AAAA", "AAAA"),
+    ]);
+    var newBlock = new MockBlock(0x7778, "BBBB", "AAAA");
+    var blockRetention = 5;
+
+    reconcileBlockHistoryEs5(getBlockByHash, blockHistory, newBlock, onBlockAdded, onBlockRemoved, blockRetention, function (error, newHistory) {
+      expect(error).to.be.undefined;
+      expect(newHistory.toJS()).to.deep.equal([
+        new MockBlock(0x7777, "AAAA", "AAAA"),
+        new MockBlock(0x7778, "BBBB", "AAAA"),
+      ]);
+      expect(newBlockAnnouncements).to.deep.equal([new MockBlock(0x7778, "BBBB", "AAAA")]);
+      expect(blockRemovalAnnouncments).to.deep.equal([new MockBlock(0x7778, "AAAA", "AAAA")]);
+      done();
+    });
+  });
+});
+
+describe("reconcileBlocksAndLogsEs5", function () {
+  var logAddedAnnonucements;
+  var logRemovedAnnouncements;
+  function onLogAdded(block, callback) { setTimeout(function () { logAddedAnnonucements.push(block); callback(); }); }
+  function onLogRemoved(block, callback) { setTimeout(function () { logRemovedAnnouncements.push(block); callback(); }); }
+
+  beforeEach(() => {
+    logAddedAnnonucements = [];
+    logRemovedAnnouncements = [];
+  });
+
+  it("generally works", function (done) {
```

Comment:
This test is probably of the most interest to you @tinybike, it is a general functionality test of a reorg+backfill using the simplified interface (blocks & logs together) and the ES5 callback wrapper (no promises).